consensus/consensus-types/src/block_info.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::block::Block;
+use crypto::hash::HashValue;
+use failure::prelude::{Error, Result};
+use libra_types::transaction::Version;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+/// The round of a block is a consensus-internal counter, which starts with 0 and increases
+/// monotonically. It is used for the protocol safety and liveness (please see the detailed protocol description).
+pub type Round = u64;
+
+/// This structure contains all the information needed for tracking a block
+/// without having access to the block or its execution output state. It
+/// assumes that the block is the last block executed within the ledger.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct BlockInfo {
```

Comment:
Fine with that. I went with alphabetical, but contextual works as well. Curious about how you arrived at this contextual ordering. For instance, I would imagine epoch_num, id, round, timestamp, executed_state_id, version but cannot determine your approach.

Reply:
Uh got it (I was wondering about your choice of the order but didn't think about the alphabetical one).
As I said above, it's really subjective, so it's up to you.
My ordering would follow the description of the block metadata:
First, you specify what epoch the block belongs to, so `epoch_num` goes first.
`round` goes next because it specifies the slot of the proposal within the epoch.
`id` goes next specifies the details of the proposal
`executed_state_id` and `version` specify the result of the execution of that proposal
`timestamp_usecs` goes last because I couldn't find a better place :P.

Reply:
Thanks for the insights! My preference is to match the original authors' style(s) as much as possible. Code like a book should ideally appear to have been written by one author.